### PR TITLE
fix plugin metadata not in results

### PR DESCRIPTION
**Pull request by SePHTaN**
another fix to tag sources case, causing plugin metadata not in results
convert all tagsources to lowercase, also occurrences of 'getID3'
only loop over plugins in _get_plugin_tags
return artists as array, don't prepare metadata outside VaInfo
HINT: replaygain metadata partly broken

### CODE REVIEW

File: src/Module/Util/VaInfo.php [hunk 1]
```diff
@@ -401,7 +402,7 @@ public static function get_tag_type($results, $configKey = 'metadata_order')
             'getid3_tag_order' => static::getConfigContainer()->get(ConfigurationKeyEnum::GETID3_TAG_ORDER)
         ];
 
-        $order = (array) ($tagorderMap[$configKey] ?? '');
+        $order = array_map('strtolower', $tagorderMap[$configKey] ?? '');
```
Comment (lachlan-00) [inline src/Module/Util/VaInfo.php:405]: for array_map should the empty string be an empty array instead?

File: src/Module/Util/VaInfo.php [hunk 2]
```diff
@@ -401,7 +402,7 @@ public static function get_tag_type($results, $configKey = 'metadata_order')
             'getid3_tag_order' => static::getConfigContainer()->get(ConfigurationKeyEnum::GETID3_TAG_ORDER)
```
Comment (lachlan-00) [inline src/Module/Util/VaInfo.php:402]: does static::getConfigContainer()->get return arrays or strings?

**Comment by lachlan-00** [comment]
just these issues i can see in scrutinizer
https://scrutinizer-ci.com/g/ampache/ampache/inspections/3de017b3-e20c-492f-a457-5d14bb28b313/issues/files/src/Module/Util/VaInfo.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

File: src/Module/Util/VaInfo.php (hunk 2 above)
Comment (usox) [reply src/Module/Util/VaInfo.php:402]: @lachlan-00 Should be the same as AmpConfig::get.

File: src/Module/Util/VaInfo.php [hunk 3]
```diff
@@ -757,7 +760,8 @@ private function get_metadata_order()
...
         ];
 
-        return (array) ($tagorderMap[$this->get_metadata_order_key()] ?? '');
+        // convert to lower case to be sure it matches plugin names in Ampache\Plugin\PluginEnum
+        return array_map('strtolower', $tagorderMap[$this->get_metadata_order_key()] ?? '');
```
Comment (usox) [inline src/Module/Util/VaInfo.php:764]: Any chance for multibyte characters? mb_strtolower would be better then

File: src/Module/Util/VaInfo.php (hunk 1 above)
Comment (SePHTaN) [reply src/Module/Util/VaInfo.php:405]: interesting finding even for me who produced it
$configKey can only be one of the three : metadata_order, metadata_order_video, getid_tag_order
these are hardcoded so if nobody produces a typo there is no chance $tagorderMap[$configKey] is undefined
$order = array_map('strtolower', $tagorderMap[$configKey]); should be sufficient
the other way 
$order = array_map('strtolower', $tagorderMap[$configKey] ?? []);
would presume that something might go wrong and than additional checks would be advidsed

File: src/Module/Util/VaInfo.php (hunk 2 above)
Comment (SePHTaN) [reply src/Module/Util/VaInfo.php:402]: both return arrays:
example for metadata_oder
$order : array (
  0 => 'getid3',
  1 => 'musicbrainz',
  2 => 'theaudiodb',
  3 => 'filename',
)

File: src/Module/Util/VaInfo.php (hunk 1 above)
Comment (usox) [reply src/Module/Util/VaInfo.php:405]: I'd say, `array_map('strtolower', $tagorderMap[$configKey] ?? [])` should be sufficient.

File: src/Module/Util/VaInfo.php (hunk 2 above)
Comment (lachlan-00) [reply src/Module/Util/VaInfo.php:402]: thank you for confirming

File: src/Module/Util/VaInfo.php (hunk 3 above)
Comment (lachlan-00) [reply src/Module/Util/VaInfo.php]: there isn't now and i think unlikely will so not a big deal

**Review by lachlan-00 — APPROVED** [review]

**Pull request merged**